internal/pkg/cli/deploy/env.go

```diff
@@ -160,7 +160,7 @@ func (d *envDeployer) validateCDN(mft *manifest.Environment) error {
 		}
 	}
 
-	if mft.CDNEnabled() && mft.CDNDoesTLSTermination() && (mft.HasImportedPublicALBCerts() || d.app.Domain != "") {
+	if mft.CDNEnabled() && mft.CDNDoesTLSTermination() {
```

Comment:
qq: Why do we no longer need the second part?

Reply:
good question! we want to allow people to use CDN TLS termination without importing public alb certs or having a copilot managed domain. So they could have just the CDN cert and use aliases from that.

As part of this change, we also had to remove the error [here](https://github.com/aws/copilot-cli/pull/4061/files#diff-924fba6d7ac2c8e2b099ec2776e78f958c7b225d5484001289b0e62283df3db1R241-R242)

Reply:
I think - iirc -  the redirection rule is created only when HTTPS is enabled => only one public ALB cert is created / app.domain != nil. Wasn't that why we needed that second part back then (and maybe now as well)?

Reply:
So basically it is possible for service to not enable HTTPS at all (native domain or import cert) and in that scenario by updating the function that check if HTTP redirect is enabled or not to not throw any error for HTTP service.

Reply:
I guess my point is just that if https is not enabled at all, there is no point to validate if any redirect is there. But yeah it doesn't hurt to check.

Reply:
That is true. I think we should return early and don't call `validateALBWorkloadsDontRedirect` at all.

Reply:
so the problem with returning early is just because someone has updated their environment manifest to not have public alb certs, we can't assume that they haven't deployed https services and are now _moving_ the cert to the cdn. in that case we still want to tell them to disable the redirect on those services.

additionally, as part of this PR, customers can deploy "HTTPS" services (meaning we create two listener rules) without importing any http public certs and only relying on the cdn cert. then as people make other manifest changes and deploy, it's still relevant to check and warn them if some services are redirecting.